### PR TITLE
KIALI-1822 Migrate istio (badging) appender to business layer API

### PR DESCRIPTION
**Pull request by jshaughn**
- move some supporting logic/tests from istio_details to business
- remove some unused functions from istio_details_service
  - CheckVirtualServiceSubset
  - FilterByDestination
- add appender test file

### CODE REVIEW

**Comment by rhqci** [comment]
:heavy_check_mark: Jenkins CI: kiali-core-pr-e2e-test [#252](https://paste.fedoraproject.org/paste/GEczG8hxJ2wJqLictf8~mQ)
  * :heavy_check_mark: Istio-and-Kiali-on-Openshift [#[1760]](https://paste.fedoraproject.org/paste/9LlSVTtJfivU-LfTcadL7w)
  * :heavy_check_mark: Istio-Upstream [#[606]](https://paste.fedoraproject.org/paste/Xk87~rJCQzO3pysy9swtAQ)
  * :heavy_check_mark: Kiali-Istio-Checker [#[2067, 2068]](https://paste.fedoraproject.org/paste/iyoa9u~HQ7TILYY~7Bu5CQ)
  * :heavy_check_mark: install-bookinfo [#[1136]](https://paste.fedoraproject.org/paste/ErhaAo6WpD8GemlwDvoz6w)
  * :heavy_check_mark: Kiali-on-Openshift [#[1126]](https://paste.fedoraproject.org/paste/cc1H5tyysfPKlXd85o-AkQ)
  * :heavy_check_mark: run-kiali-e2e-tests [#[532]](https://paste.fedoraproject.org/paste/E8fJwNByMX~6R91vgn2yCg)


KIALI_NAME=kialicorepr590-istio-system

File: models/virtual_service.go [hunk 1]
```diff
@@ -56,3 +56,18 @@ func (vService *VirtualService) Parse(virtualService kubernetes.IstioObject) {
 	vService.Tcp = virtualService.GetSpec()["tcp"]
 	vService.Tls = virtualService.GetSpec()["tls"]
 }
+
+// IsVirtualService returns true if VirtualService hosts applies to the service
+func (vService *VirtualService) IsVirtualService(namespace string, serviceName string) bool {
```
Comment (lucasponce) [inline models/virtual_service.go]: This name is misleading.
A VirtualService model is self invoking if IsVirtualService() ? It doesn't sound good to me; instead this method is validating if a service name is part of the hosts.
Suggestion IsValidHost() sounds better.

Comment (lucasponce) [reply models/virtual_service.go]: Also, this name can be propagated into the Test naming strategy as well.

Comment (jshaughn) [reply models/virtual_service.go]: Funny, I wasn't really happy with this name either, I will change it.

**Comment by rhqci** [comment]
:heavy_check_mark: Jenkins CI: kiali-core-pr-e2e-test [#262](https://paste.fedoraproject.org/paste/LXfZBcKPGg4OQx3D4shP-w)
  * :heavy_check_mark: Istio-and-Kiali-on-Openshift [#[1781]](https://paste.fedoraproject.org/paste/YoIViH9u6qJQzQubNe5EQg)
  * :heavy_check_mark: Istio-Upstream [#[626]](https://paste.fedoraproject.org/paste/rpjdq2gJGcrf5CYPwgIsIg)
  * :heavy_check_mark: Kiali-Istio-Checker [#[2113, 2114]](https://paste.fedoraproject.org/paste/BXIkce5VhTw5RjnP5H4OhA)
  * :heavy_check_mark: install-bookinfo [#[1158]](https://paste.fedoraproject.org/paste/z17TEdCa2626RDEbwhvzJQ)
  * :heavy_check_mark: Kiali-on-Openshift [#[1154]](https://paste.fedoraproject.org/paste/CO-Co1dfZRunpNVgVC6xnA)
  * :heavy_check_mark: run-kiali-e2e-tests [#[556]](https://paste.fedoraproject.org/paste/GGpwobWH-AkmHAHtn7kJgA)


KIALI_NAME=kialicorepr590-istio-system

**Pull request merged**